detect_secrets/core/secrets_collection.py

```diff
@@ -251,18 +251,48 @@ def _results_accumulator(self, filename):
                  Caller is responsible for updating the dictionary with
                  results of plugin analysis.
         """
-        results = {}
+        file_results = {}
 
         for plugin in self.plugins:
-            yield results, plugin
+            yield file_results, plugin
 
-        if not results:
+        if not file_results:
             return
 
+        self._remove_password_secrets_if_line_reported_already(
+            file_results,
+        )
+
         if filename not in self.data:
-            self.data[filename] = results
+            self.data[filename] = file_results
         else:
-            self.data[filename].update(results)
+            self.data[filename].update(file_results)
+
+    def _remove_password_secrets_if_line_reported_already(
```

Comment:
Instead of removing, why don't we change our baseline schema to display types in list form? This way, we can preserve all the intelligence that we gather on a particular flagged secret.

Reply:
Well, it would break all the existing baseline's we made ([like when we changed the high entropy string types](https://github.com/Yelp/detect-secrets/pull/26/files)). I'm not sure how hard that would be to implement though, I can look into that.

Reply:
Hmm, good point. Though, arguably a step in the right direction?

Thankfully, we built things for doing better version bumps, since the last time. We would need to update the [merge_baseline](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/core/baseline.py#L151) function, and the pre-commit hook should get all *active* clients [slowly upgrading](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/pre_commit_hook.py#L162).

The only gotchas I see are:
* Server needs to be updated (and handle stale clients through detected baseline versions)
* We need to handle this case in [audit](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/core/audit.py#L315).

Perhaps the fix is merely to implement the dedup logic in auditing (as it's the only place that I can find that uses the `type` output from the baseline)?

e.g.

```
def to_list(value):
    try:
        iter(value)
        return value
    except ValueError:
        return [value]

...

plugin = initialize.from_secret_type(
    to_list(secret['type']),
    plugin_settings,
)
```

Reply:
When you said `Perhaps the fix is merely to implement the dedup logic in auditing`, did you mean you don't mind having all the `Password` ones separate from the others in the baseline? We might double our baseline sizes but I think I'm leaning towards that a little :D

Reply:
(Stating for the record, discussed in person yesterday, that we will just take the baseline size increase, and punt of de-duping in the audit functionality until later.)